slingr-stack/proxy-service
# Migrate proxy endpoint to service

Pull-request for issue #[14883](https://github.com/slingr-stack/platform/issues/14883) created by @agreggio-slingr 

## What it does?

Migrate endpoint proxy to service

## How to test it?

It requires having ngrok installed and configured on port 10001.

`ngrok http 10001`

Install this [proxy-service-text](https://github.com/slingr-stack/proxy-service-test) locally in the developer portal, you will need to perform the following proxy-service configuration in the builder:

![Screenshot from 2023-10-24 15-13-07](https://github.com/slingr-stack/proxy-service/assets/110241411/50728839-6e83-4c7c-aaca-d4fb7587e5f7)

In your service_local.properties file, you should configure the serviceUri and serviceToken generated in the builder.

Once the service is installed, you should copy the "Configuration" section in your .properties of service you want to test, for example for http-service:

```
# System properties
_service_name=proxy
_app_name=test1
_environment=dev
_pod_id=id
_profile=default
_custom_domain=
_debug=true
_local_deployment=true
_base_domain=localhost:8000
_webservices_port=10001
_extension_broker_api=http://test1.localhost:8000/dev/services/proxy/api
_token=9a6adbc4-d714-492e-90cc-25fd908f71f0

# Service specific properties
#---------------------
# no auth
#---------------------
_service_config={"baseUrl":"https://postman-echo.com", "defaultHeaders":"", "emptyPath":"", "rememberCookies":"true", "allowExternalUrl":"false", "connectionTimeout":"1004", "readTimeout":"1504", "followRedirects":"false", "authType":"", "username":"", "password":""}
```

Finally, start the proxy first, then the service to test, and execute some of the service methods, for example, for the http-service

`svc.proxy.get({path:"/get?foo1="});`

## Implementation notes

N/A

## Deployment notes

N/A
